examples/passport-login/src/application.ts

```diff
@@ -51,11 +75,39 @@ export class OAuth2LoginApplication extends BootMixin(
     this.bind(UserServiceBindings.PASSPORT_USER_IDENTITY_SERVICE).toClass(
       PassportUserIdentityService,
     );
+    // passport strategies
+    this.add(createBindingFromClass(FacebookOauth, {key: 'facebookStrategy'}));
+    this.add(createBindingFromClass(GoogleOauth, {key: 'googleStrategy'}));
+    this.add(createBindingFromClass(CustomOauth2, {key: 'oauth2Strategy'}));
+    // passport express middleware
+    this.add(
+      createBindingFromClass(FacebookOauth2ExpressMiddleware, {
+        key: 'facebookStrategyMiddleware',
+      }),
+    );
+    this.add(
+      createBindingFromClass(GoogleOauth2ExpressMiddleware, {
+        key: 'googleStrategyMiddleware',
+      }),
+    );
+    this.add(
+      createBindingFromClass(CustomOauth2ExpressMiddleware, {
+        key: 'oauth2StrategyMiddleware',
+      }),
+    );
+    // LoopBack 4 style authentication strategies
     this.add(createBindingFromClass(LocalAuthStrategy));
     this.add(createBindingFromClass(FaceBookOauth2Authorization));
     this.add(createBindingFromClass(GoogleOauth2Authorization));
     this.add(createBindingFromClass(Oauth2AuthStrategy));
     this.add(createBindingFromClass(SessionStrategy));
     this.add(createBindingFromClass(BasicStrategy));
+    // Express style middleware interceptors
+    this.bind('passport-init-mw').to(toInterceptor(passport.initialize()));
+    this.bind('passport-session-mw').to(toInterceptor(passport.session()));
+    this.bind('passport-facebook').toProvider(FacebookOauthInterceptor);
+    this.bind('passport-google').toProvider(GoogleOauthInterceptor);
+    this.bind('passport-oauth2').toProvider(CustomOauth2Interceptor);
+    this.bind('set-session-user').toProvider(SessionAuth);
```

Comment:
We should be able to make these interceptor providers singleton too.

Reply:
agreed